src/scene/container/Container.ts

```diff
@@ -852,7 +855,64 @@ export class Container<C extends ContainerChild = ContainerChild> extends EventE
         return this._worldTransform;
     }
 
-    // / ////// transform related stuff
+    /**
+     * Returns the global transform matrix of the container within the scene.
+     * @param matrix - Optional matrix to store the result. If not provided, a new Matrix will be created.
+     * @param skipUpdate - Performance optimization flag:
+     *   - If false (default): Recalculates the entire transform chain for accuracy
+     *   - If true: Uses cached worldTransform from the last render pass for better performance
+     * @returns The resulting transformation matrix (either the input matrix or a new one)
+     * @example
+     * // Accurate but slower - recalculates entire transform chain
+     * const preciseTransform = container.getGlobalTransform();
+     *
+     * // Faster but may be outdated - uses cached transform
+     * const cachedTransform = container.getGlobalTransform(undefined, true);
+     *
+     * // Reuse existing matrix
+     * const existingMatrix = new Matrix();
+     * container.getGlobalTransform(existingMatrix);
+     */
+    public getGlobalTransform(matrix?: Matrix, skipUpdate?: boolean): Matrix
```

Comment:
What do you think about this arguments order? Would skipUpdate then matrix make more or less sense? I'm not sure I have a POV but am asking. If both args are equally optional, then options may be better choice.

Reply:
I was following the getGlobalBounds sig - but would be open to changing them all to be option objects if we think thats a better direction! 

I would like to encourage devs to pass in a matrix - so that a new one is not created each time the function is called.

Reply:
I am okay with this I think unless @Zyie has any other thoughts.